pkg/controller/datavolume-controller.go

```diff
@@ -438,6 +438,20 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		return reconcile.Result{}, err
 	}
 
+	// When cloning a PVC, if the target's size is not specified,
+	// said value can be attainable from the source PVC
+	if selectedCloneStrategy != NoClone {
+		targetRequest, hasTargetRequest := pvcSpec.Resources.Requests[corev1.ResourceStorage]
+		if hasTargetRequest && targetRequest.IsZero() {
+			sourcePvc, _ := r.findSourcePvc(datavolume)
+			if selectedCloneStrategy == SmartClone || selectedCloneStrategy == CsiClone {
+				pvcSpec.Resources.Requests[corev1.ResourceStorage] = *sourcePvc.Status.Capacity.Storage()
+			} else {
+				// HostAssistedClone
```

Comment:
will there be a something here or is it supposed to be like this? what happens in this case?

Reply:
Yeah, still in progress. That's probably the hardest case so I'll still need some time before committing the changes.